### PR TITLE
 Model API for domain elements

### PR DESCRIPTION
**Pull request by ajreynol**
This adds an API call from theory_model / model to get the domain elements of an uninterpreted sort.  It updates the smt2 printer to use this call.

@barrettcw , this is relevant to the discussion of whether CVC4 can reparse its own models (it is a step towards this).

The new API will have to call into TheoryModel::getDomainElements to add this feature.

### CODE REVIEW

**Comment by ajreynol** [comment]
This fixes #2851.

File: src/smt/model.h [hunk 1]
```diff
@@ -96,6 +96,14 @@ class Model {
    * is a predicate over t that indicates a property that t satisfies.
    */
   virtual std::vector<std::pair<Expr, Expr> > getApproximations() const = 0;
+  /** get the domain elements for uninterpreted sort s
+   *
+   * This method is gets the interpretation of an uninterpreted sort s.
```
Comment (aniemetz) [inline src/smt/model.h]:
```suggestion
   * This method gets the interpretation of an uninterpreted sort s.
```

Comment (aniemetz) [reply src/smt/model.h]: Also, what is `uninterpreted sort s` here -- the given `Type t`? Maybe make this more clear?

Comment (ajreynol) [reply src/smt/model.h]: Yes, meant to say `t`.

**Pull request merged**